main.go

```diff
@@ -332,9 +347,197 @@ func mainCore() error {
+	displaySyncStatusPage := blocksBehind > cfg.SyncStatusLimit ||
+		(usePG && auxDBheight < 40000) || wireDBheight < 40000
+
+	// Set the displaySyncStatusPage value.
+	explore.SetDisplaySyncStatusPage(displaySyncStatusPage)
+
+	// Initiate the sync status monitor and SyncStatusUpdate goroutine if the
+	// sync status is activated or else initiate the goroutine that handles
+	// storing blocks needed for the explorer pages.
+	if displaySyncStatusPage {
+		// Starts a goroutine that signals the websocket to check and send to
+		// the frontend the latest sync status progress updates.
+		explore.StartSyncingStatusMonitor()
+	} else {
+		// Set that blocks freshly sync'd to to be stored for the explorer pages
+		// till the sync is done.
+		explorer.SetSyncExplorerUpdateStatus(true)
+
+		// This goroutines updates the blocks needed on the explorer pages. It
+		// only runs when the status sync page is not the default page that a user
+		// can view on the running webserver but the syncing of blocks behind the
+		// best block is happening in the background. No new blocks monitor from
+		// dcrd will be initiated until the best block from dcrd is in sync with
+		// the best block from dcrdata.
+		go func() {
+			for hash := range latestBlockHash {
+				// Checks if updates should be sent to the explorer. If its been
+				// deactivated it breaks the loop.
+				if !explorer.SyncExplorerUpdateStatus() {
+					break
+				}
+
+				// Setch the blockdata using its block hash.
+				d, msgBlock, err := collector.CollectHash(hash)
+				if err != nil {
+					log.Warnf("failed to fetch blockdata for (%s) hash. error: %v",
+						hash.String(), err)
+					continue
+				}
+
+				// Store the blockdata fetch for the explorer pages
+				if err = explore.Store(d, msgBlock); err != nil {
+					log.Warnf("failed to store (%s) hash's blockdata for the explorer pages error: %v",
+						hash.String(), err)
+				}
+			}
+		}()
+
+		// Stores the first block in the explorer. It should correspond with the
+		// block that is currently in sync in all the dcrdata dbs
+		loadHeight := wireDBheight
+		if usePG && (auxDBheight < wireDBheight) {
+			loadHeight = auxDBheight
+		}
+
+		// Fetches the first block hash whose blockdata will be loaded in the
+		// explorer for it pages.
+		loadBlockHash, err := dcrdClient.GetBlockHash(int64(loadHeight))
+		if err != nil {
+			return fmt.Errorf("failed to fetch the block at height (%d), dcrdata dbs must be corrupted",
+				loadHeight)
+		}
+
+		// Signal the goroutine to load this block hash's data.
+		latestBlockHash <- loadBlockHash
+	}
+
+	// Starts a goroutine that fetches the raw updates, process them and
+	// set them as the latest sync status progress updates. The goroutine exits
+	// when no sync is running.
+	explore.SyncStatusUpdate(barLoad)
+
 	blockDataSavers = append(blockDataSavers, explore)
 
-	// Sync up with the blockchain
+	// Register for notifications from dcrd
+	cerr := notify.RegisterNodeNtfnHandlers(dcrdClient)
```

Comment:
This should still be move way down near `initiateHandlersAndCollectBlocks`.  With this here dcrdata unnecessarily gets notifications for new blocks and transactions, although it has not yet launched any of the monitors (that's what the goroutines in `initiateHandlersAndCollectBlocks` do).

The handlers defined in `MakeNodeNtfnHandlers` and subsequently used when creating the RPC client just shuffle the data to the monitors, which are started in `initiateHandlersAndCollectBlocks`.  Note the design of the `rpcclient.NotificationHandlers` defined in `notification.MakeNodeNtfnHandlers`, and how the generally do not hang if the monitor is not receiving on the channel (the select-case-default).

The one handler that does send successfully is `OnBlockConnected` this is because the `(*collectionQueue).ProcessBlocks()` is always started, just with no synchronous handler functions defined for it to use (they get set via `collectionQueue.SetSynchronousHandlers` in `initiateHandlersAndCollectBlocks`).  This is evident with debug logging turned on as we see the following kind of message through the entire process:

    [DBG] NTFN: Synchronous handlers of collectionQueue.ProcessBlocks() completed in 242ns

While all of the above seems mostly harmless, it is not because a large number of the channel sends will "succeed" in filling up the channel buffer, only to be pulled out of the buffer loooong after the main sync is done.  When they do get pulled out of the buffer, these blocks and transactions are all already processed.  They should not be processed again.

For these reasons, it is necessary to not register for notifications from dcrd until we are ready (sync and indexing are completed).